protoc-gen-go/generator/generator.go

```diff
@@ -655,42 +655,41 @@ func (g *Generator) CommandLineParameters(parameter string) {
 // If its file is in a different package, it returns the package name we're using for this file, plus ".".
 // Otherwise it returns the empty string.
 func (g *Generator) DefaultPackageName(obj Object) string {
-	importPath := obj.ImportPath()
+	importPath := obj.GoImportPath()
 	if importPath == g.file.importPath {
 		return ""
 	}
-	return string(g.PackageName(importPath)) + "."
+	return string(g.GoPackageName(importPath)) + "."
 }
 
-// PackageName returns the name used for a package.
-func (g *Generator) PackageName(importPath ImportPath) PackageName {
+// GoPackageName returns the name used for a package.
+func (g *Generator) GoPackageName(importPath GoImportPath) GoPackageName {
 	if name, ok := g.packageNames[importPath]; ok {
 		return name
 	}
 	name := cleanPackageName(baseName(string(importPath)))
 	for i, orig := 1, name; g.usedPackageNames[name]; i++ {
-		name = orig + PackageName(strconv.Itoa(i))
+		name = orig + GoPackageName(strconv.Itoa(i))
 	}
 	g.packageNames[importPath] = name
 	g.usedPackageNames[name] = true
 	return name
 }
 
-var globalPackageNames = map[PackageName]int{
-	"fmt":   1,
-	"math":  1,
-	"proto": 1,
+var globalPackageNames = map[GoPackageName]bool{
+	"fmt":   true,
+	"math":  true,
+	"proto": true,
 }
 
 // Create and remember a guaranteed unique package name. Pkg is the candidate name.
 // The FileDescriptor parameter is unused.
 func RegisterUniquePackageName(pkg string, f *FileDescriptor) string {
 	name := cleanPackageName(pkg)
-	globalPackageNames[name]++
-	if n := globalPackageNames[name]; n > 1 {
-		return pkg + strconv.Itoa(n)
+	for i, orig := 1, name; globalPackageNames[name]; i++ {
+		name = orig + GoPackageName(strconv.Itoa(i))
```

Comment:
Don't you still have to mutate the `globalPackageNames` eventually?

Reply:
Ack. Done.

This should have been caught by a test, but takes a lot of infrastructure to test. Do you want one?

Reply:
This is only to support plugins, so whatever. We can add a test in the future.